lib/srtm.py

```
@@ -0,0 +1,385 @@
+    parsed_path = Path(urlparse(url).path)
+    download_path = (
+        cache_dir / f"{tile_name}{''.join(parsed_path.suffixes) or '.download'}"
+    )
+    partial_hgt_path = cache_dir / f"{tile_name}.hgt.tmp"
+    direct_hgt_download = download_path == hgt_path
+    if direct_hgt_download:
+        download_path = partial_hgt_path
+
+    try:
+        download_path.unlink(missing_ok=True)
+        with urlopen(url, timeout=60) as response, download_path.open("wb") as out:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)